### PR TITLE
boards: added ble400

### PR DESCRIPTION
**Pull request by sdalu**
BLE400 board (nRF51 cpu, default to ac model)

### CODE REVIEW

File: boards/ble400/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# define the used CPU
+export CPU = nrf51
+export CPU_MODEL = nrf51x22xxac
+
+# define the default port depending on the host OS
+PORT_LINUX ?= /dev/ttyUSB0
```
Comment (PeterKietzmann) [inline boards/ble400/Makefile.include]: Not ACM? Didn't check it. Just referring to the nrf51 dongle

File: boards/ble400/dist/openocd.cfg [hunk 2]
```diff
@@ -0,0 +1,5 @@
+source [find interface/jlink.cfg]
+transport select swd
```
Comment (PeterKietzmann) [inline boards/ble400/dist/openocd.cfg]: Honestly speaking I have no experience in OpenOCD configs but comparing to other nrf51 based boards it it `hla_swd`. Can you give me a quick hint about that line please?

File: boards/ble400/include/board.h [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+#endif
+
+/**
+ * @brief   Xtimer configuration
+ * @{
+ */
+#define XTIMER                      (0)
+#define XTIMER_CHAN                 (0)
+#define XTIMER_MASK                 (0xff000000)
+#define XTIMER_SHIFT_ON_COMPARE     (2)
```
Comment (PeterKietzmann) [inline boards/ble400/include/board.h]: Not needed anymore after #5310

File: boards/ble400/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * @name UART configuration
+ * @{
+ */
+#define UART_NUMOF          (1U)
+/* UART pin configuration */
+#define UART_HWFLOWCTRL     1
+#define UART_PIN_RX         11
+#define UART_PIN_TX         9
+#define UART_PIN_RTS	    8
```
Comment (PeterKietzmann) [inline boards/ble400/include/periph_conf.h]: I know, it's picky but would you add a space to fix the indentation here :-)

File: boards/ble400/Makefile.include (hunk 1 above)
Comment (sdalu) [inline boards/ble400/Makefile.include]: No ACM

File: boards/ble400/dist/openocd.cfg (hunk 2 above)
Comment (sdalu) [inline boards/ble400/dist/openocd.cfg]: Neither do I. Both `swd` and `hla_swd` seem to work. I can change it to `hla_swd`

File: boards/ble400/include/board.h (hunk 3 above)
Comment (sdalu) [inline boards/ble400/include/board.h]: What is not needed? The whole XTIMER section or only the XTIMER_SHIFT_ON_COMPARE line?

Comment (PeterKietzmann) [inline boards/ble400/include/board.h]: Just `XTIMER_SHIFT_ON_COMPARE` is not needed

File: boards/ble400/dist/openocd.cfg (hunk 2 above)
Comment (PeterKietzmann) [inline boards/ble400/dist/openocd.cfg]: Will try to find out. Tomorw, have to leave now..

**Comment by PeterKietzmann** [comment]
The code looks fine for me but don't have the hardware. @cgundogan do you have access to the board? Maybe on Hauke's desk or so?

File: boards/ble400/include/board.h (hunk 3 above)
Comment (sdalu) [inline boards/ble400/include/board.h]: Done

**Comment by PeterKietzmann** [comment]
Thanks. IMO you can already squash your second commit

**Comment by sdalu** [comment]
I'm not that good with git,  I'll let you do the "squashing"

**Comment by PeterKietzmann** [comment]
Hi @sdalu I'm sorry but this must be done in your local repo before pushing. See section [How to squash commits](https://github.com/RIOT-OS/RIOT/wiki/Git-cheatsheet) for some help about squashing. If you think you can't handle it, just give a sign. Here are some experts around (not talking about myself)

**Comment by sdalu** [comment]
Guess what... I made things worst.

**Comment by PeterKietzmann** [comment]
@sdalu I took your commit and opened a new PR (#5448) so you don't have to mess up with the git history now. Please have a look at it! Thanks for your great work! Next time please follow our contribution guidelines as described [here](https://github.com/RIOT-OS/RIOT/wiki/Contributing-to-RIOT) and if needed the [git cheatsheet](https://github.com/RIOT-OS/RIOT/wiki/Git-cheatsheet)

**Pull request closed without merging**